<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="input" default="filtered_cloud" doc="Input cloud topic."/>
    <arg name="output" default="corrected_cloud" doc="Output cloud topic."/>
    <arg name="publish_filtered" default="$(eval bool(filtered_cloud))"/>
    <arg name="nn_r" default="0.2" doc="Nearest neighbor radius."/>
    <arg name="eigenvalue_bounds" default="[[0, -.inf, 0.0004], [1, 0.0025, .inf]]"
         doc="Eigenvalue bounds for correcting depth."/>
    <arg name="model_class" default="ScaledPolynomial" doc="Class name from module depth_correction.model."/>
    <arg name="model_state_dict" default="$(find depth_correction)/config/weights/ScaledPolynomial.pth"
         doc="Path to the model state dict."/>

    <node name="depth_correction" pkg="depth_correction" type="depth_correction" output="screen">
        <rosparam subst_value="true">
            model:
                class: $(arg model_class)
                state_dict: '$(arg model_state_dict)'
            device: cpu
            float_type: float64
            max_age: 1.0
            publish_filtered: $(arg publish_filtered)
            nn_r: $(arg nn_r)
            eigenvalue_bounds: $(arg eigenvalue_bounds)
        </rosparam>
        <remap from="cloud" to="$(arg cloud)"/>
        <remap from="corrected_cloud" to="$(arg corrected_cloud)"/>
    </node>
</launch>
